### PR TITLE
No implicit conversions of Task to Instance during tests

### LINKED ISSUE #4428: UnitTests: use Instances instead of Tasks

### PR DESCRIPTION
**Pull request by unterstein**
Introduced TestInstanceBuilder and TestTaskBuilder to
- remove implicit conversion
- remove most usage of Instance(task)
- remove InstanceSupport trait

Tests and IntegrationTests pass locally!
Fixes #4428

### CODE REVIEW

**Comment by unterstein** [comment]
Fixes #4428

File: src/test/scala/mesosphere/marathon/builder/InstanceBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import mesosphere.marathon.core.instance.update.InstanceUpdateOperation
+import mesosphere.marathon.core.instance.{ Instance, InstanceStatus, TestTaskBuilder }
+import mesosphere.marathon.core.pod.MesosContainer
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.state.{ PathId, Timestamp }
+import org.apache.mesos
+
+import scala.collection.immutable.Seq
+
+case class InstanceBuilder(
```
Comment (meichstedt) [inline src/test/scala/mesosphere/marathon/builder/InstanceBuilder.scala]: What is this class used for? Seems like a dupe of TestInstanceBuilder

File: src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryHelper.scala [hunk 2]
```diff
@@ -18,14 +18,14 @@ class InstanceOpFactoryHelper(
 
   def launchEphemeral(
     taskInfo: Mesos.TaskInfo,
-    newTask: Task.LaunchedEphemeral): InstanceOp.LaunchTask = {
+    newTask: Task.LaunchedEphemeral,
+    instance: Instance): InstanceOp.LaunchTask = {
```
Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryHelper.scala]: Would be nice to get rid of `newTask`, but for the sake of this PR it's ok.

File: src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala [hunk 3]
```diff
@@ -92,7 +92,7 @@ class InstanceOpFactoryImpl(
             hostPorts = ports.flatten
           )
 
-          taskOperationFactory.launchEphemeral(taskInfo, task)
+          taskOperationFactory.launchEphemeral(taskInfo, task, Instance(task)) // TODO PODS replace Instance(task)
```
Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala]: Nit: you could as well build the instance using its default ctor above and get rid of using the custom ctor right now. Not necessary for this PR though.

**Comment by meichstedt** [comment]
2 unrelated shaky tests, passed locally – merging.

**Pull request merged**